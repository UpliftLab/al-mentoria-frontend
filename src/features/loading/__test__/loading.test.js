import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Loading from '../loading';

describe('Loading', () => {
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer
      .create(<Loading />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the green border', () => {
    render(
      <Loading />,
    );

    const loading = screen.getByTestId('loading');
    const { classList } = loading;

    expect(classList).toContain('border-t-lime-500');
  });
});
