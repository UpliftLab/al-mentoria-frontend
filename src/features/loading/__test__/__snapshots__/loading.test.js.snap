// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly in the DOM 1`] = `
<span
  className="h-6 w-6 block rounded-full border border-t-2 border-t-lime-500 animate-spin"
  data-testid="loading"
/>
`;
